typewise_alert.py

```diff
@@ -1,46 +1,33 @@
+from collections import defaultdict 
 
 def infer_breach(value, lowerLimit, upperLimit):
-  if value < lowerLimit:
-    return 'TOO_LOW'
-  if value > upperLimit:
-    return 'TOO_HIGH'
-  return 'NORMAL'
-
+    if value < lowerLimit:
+        return 'TOO_LOW'
+    elif value > upperLimit:
+        return 'TOO_HIGH'
+    return 'NORMAL'
 
 def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
+    coolingTypes = ['PASSIVE_COOLING', 'HI_ACTIVE_COOLING', 'MED_ACTIVE_COOLING']
+    upperLimit = [35, 45, 40]
+    return infer_breach(temperatureInC, 0, upperLimit[coolingTypes.index(coolingType)])\
+if coolingType in coolingTypes else infer_breach(temperatureInC, 0, 0)
 
+def def_value():
+    return "Not Present"
 
 def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
+    breachType = classify_temperature_breach(batteryChar['CoolingType'], temperatureInC)
+    d = defaultdict(def_value)
+    d["TO_CONTROLLER"] = send_to_controller(breachType)
```

Comment:
Is the intention to store function pointer or call the function.
this looks like both send functions are called and return values are stored in dictionary.
Please correct me if I am wrong.